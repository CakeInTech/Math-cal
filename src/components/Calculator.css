* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

.main {
  display: flex;
  flex-direction: row;
  /* align-items: center; */
  padding-top: 70px;
  justify-content: space-around;
}

.min-Calculator {
  width: 50%;
}

h1 {
  width: 50%;
  font-size: 35px;
  animation-name: example;
  animation-duration: 4s;
  animation-iteration-count: infinite;
}

@keyframes example {
  0% {color: #F0FF42;}
  20% {color: #B3FFAE}
  50% {color: #9fef00;}
}

.one,
.two,
.three,
.four,
.five {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
}

.colored {
  background-color: #9fef00;
}

input[type='button'] {
  border: 0.0625rem solid #222;
  padding: 0.625rem 1.25rem;
  color: black;
  text-align: center;
  text-decoration: none;
  font-size: 1.0625rem;
}

input[type='button']:active {
  background-color: #ccc;
}

.screen {
  width: 100%;
  height: 60px;
  border: 1px solid #9fef00;
  color: #fff;
  padding: 10px;
  text-align: right;
}

.special {
  grid-column: 1/ span 2;
}
