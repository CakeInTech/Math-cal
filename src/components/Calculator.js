import React from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';

class Calculator extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      total: 0,
      next: null,
      operation: null,
    };
  }

   handleClick = (event) => {
     const value = event.target;
     const answer = calculate(this.state, value);
     this.setState(answer);
   }

   showAnswer = (events) => {
     this.setState({
       total: events.target.value,
     });
   }

   render() {
     const { total, next, operation } = this.state;
     return (
       <div className="main">
         <div className="min-calculator">
           <div className="screen" onChange={this.showAnswer}>
             {total}
             {operation}
             {next}
           </div>
           <div className="button-row one">
             <input type="button" value="AC" onClick={this.handleClick} />
             <input type="button" value="+/-" onClick={this.handleClick} />
             <input type="button" value="%" onClick={this.handleClick} />
             <input type="button" value="รท" className="colored" onClick={this.handleClick} />
           </div>
           <div className="button-row two">
             <input type="button" value="7" />
             <input type="button" value="8" />
             <input type="button" value="9" />
             <input type="button" value="*" className="colored" onClick={this.handleClick} />
           </div>
           <div className="button-row three">
             <input type="button" value="4" onClick={this.handleClick} />
             <input type="button" value="5" onClick={this.handleClick} />
             <input type="button" value="6" onClick={this.handleClick} />
             <input type="button" value="-" className="colored" onClick={this.handleClick} />
           </div>
           <div className="button-row four">
             <input type="button" value="1" onClick={this.handleClick} />
             <input type="button" value="2" onClick={this.handleClick} />
             <input type="button" value="3" onClick={this.handleClick} />
             <input type="button" value="+" className="colored" onClick={this.handleClick} />
           </div>
           <div className="button-row five">
             <input type="button" value="0" className="special" onClick={this.handleClick} />
             <input type="button" value="." onClick={this.handleClick} />
             <input type="button" value="=" className="colored" onClick={this.handleClick} />
           </div>

         </div>
       </div>
     );
   }
}

export default Calculator;
