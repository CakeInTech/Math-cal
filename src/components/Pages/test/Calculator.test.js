import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../../Calculator';

it("it renders and test's against the dom snapshots", () => {
  const tree = renderer.create(<Calculator />);
  expect(tree).toMatchSnapshot();
});
