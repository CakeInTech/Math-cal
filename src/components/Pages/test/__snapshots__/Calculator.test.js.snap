// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and test's against the dom snapshots 1`] = `
<div
  className="main"
>
  <h1>
    Hey there Let's do some math
  </h1>
  <div
    className="min-calculator"
  >
    <div
      className="screen"
      onChange={[Function]}
    >
      0
    </div>
    <div
      className="button-row one"
    >
      <input
        onClick={[Function]}
        type="button"
        value="AC"
      />
      <input
        onClick={[Function]}
        type="button"
        value="+/-"
      />
      <input
        onClick={[Function]}
        type="button"
        value="%"
      />
      <input
        className="colored"
        onClick={[Function]}
        type="button"
        value="รท"
      />
    </div>
    <div
      className="button-row two"
    >
      <input
        onClick={[Function]}
        type="button"
        value="7"
      />
      <input
        onClick={[Function]}
        type="button"
        value="8"
      />
      <input
        onClick={[Function]}
        type="button"
        value="9"
      />
      <input
        className="colored"
        onClick={[Function]}
        type="button"
        value="x"
      />
    </div>
    <div
      className="button-row three"
    >
      <input
        onClick={[Function]}
        type="button"
        value="4"
      />
      <input
        onClick={[Function]}
        type="button"
        value="5"
      />
      <input
        onClick={[Function]}
        type="button"
        value="6"
      />
      <input
        className="colored"
        onClick={[Function]}
        type="button"
        value="-"
      />
    </div>
    <div
      className="button-row four"
    >
      <input
        onClick={[Function]}
        type="button"
        value="1"
      />
      <input
        onClick={[Function]}
        type="button"
        value="2"
      />
      <input
        onClick={[Function]}
        type="button"
        value="3"
      />
      <input
        className="colored"
        onClick={[Function]}
        type="button"
        value="+"
      />
    </div>
    <div
      className="button-row five"
    >
      <input
        className="special"
        onClick={[Function]}
        type="button"
        value="0"
      />
      <input
        onClick={[Function]}
        type="button"
        value="."
      />
      <input
        className="colored"
        onClick={[Function]}
        type="button"
        value="="
      />
    </div>
  </div>
</div>
`;
